vllm/entrypoints/openai/api_server.py

```diff
@@ -174,6 +174,12 @@ async def authentication(request: Request, call_next):
             raise ValueError(f"Invalid middleware {middleware}. "
                              f"Must be a function or a class.")
 
+    return app
+
+
+def run_server(args):
```

Comment:
Is it possible to make `engine` as an optional arg to this function?
```suggestion
def run_server(args, llm_engine=None)
```
This can help external applications reuse the llm engine and attach other API interfaces (like grpc) to the same llm engine. To be used with the other suggestion below

Reply:
+1, this would be useful.